src/pure.js

```diff
@@ -80,27 +86,23 @@ export class SvelteTestingLibrary {
       unmount: () => {
         this.cleanupComponent(component)
       },
-      ...getQueriesForElement(container, queries),
+      ...getQueriesForElement(baseElement, renderOptions.queries),
     }
   }
 
-  renderComponent({ target, ComponentConstructor }, options) {
-    options = { target, ...this.checkProps(options) }
-
-    if (IS_SVELTE_5)
+  renderComponent(ComponentConstructor, componentOptions) {
+    if (IS_SVELTE_5) {
       throw new Error('for Svelte 5, use `@testing-library/svelte/svelte5`')
+    }
 
-    const component = new ComponentConstructor(options)
+    const component = new ComponentConstructor(componentOptions)
 
     this.componentCache.add(component)
 
     // TODO(mcous, 2024-02-11): remove this behavior in the next major version
-    // It is unnecessary has no path to implementation in Svelte v5
-    if (!IS_SVELTE_5) {
-      component.$$.on_destroy.push(() => {
-        this.componentCache.delete(component)
-      })
-    }
+    component.$$.on_destroy.push(() => {
+      this.componentCache.delete(component)
+    })
```

Comment:
Should I just remove this now? It's a weird little bit of logic that makes it so one can call _either_ `component.$destroy` or `unmount` in a test. It feels reasonable to only provide one way - `unmount` - to do this

Reply:
I'd tend to say "nah". It's been the established behavior, and with Svelte 4 probably being eclipsed by Svelte 5 soon, there is little gain to change the behavior now.